pkg/controller/ec2/instance/controller.go

```diff
@@ -178,26 +174,10 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 			o.DisableApiTermination = r.DisableApiTermination
 		}
 
-		if r.EbsOptimized != nil {
-			o.EbsOptimized = r.EbsOptimized
-		}
-
 		if r.InstanceInitiatedShutdownBehavior != nil {
 			o.InstanceInitiatedShutdownBehavior = r.InstanceInitiatedShutdownBehavior
 		}
 
-		if r.InstanceType != nil {
-			o.InstanceType = r.InstanceType
-		}
-
-		if r.KernelId != nil {
-			o.KernelId = r.KernelId
-		}
-
-		if r.RamdiskId != nil {
-			o.RamdiskId = r.RamdiskId
-		}
-
```

Comment:
These values are used during `LateInitialize` to update the MRs spec. See https://github.com/crossplane-contrib/provider-aws/pull/2028/files#diff-a888525d6868082318c9307abc6c1d162e9647a348dd19b77f1c9695a18fc4ccR201. So these attributes cannot just be removed.

In order to prevent unnecessary AWS calls, we would need to check if the respective spec property needs late-initialization. If that is not the case, we might be able to skip it - if it is not needed for `isUpToDate`.

Reply:
I believe in the linked code (`LateInitializeInstance`) these values are actually pulled from the `RunInstances` result. My investigation below:

The function signature for `LateInitializeInstance` takes in two in:
-  `instance *types.Instance`, populated from the `DescribeInstance` call.
- `attributes *ec2.DescribeInstanceAttributeOutput`. The `attributes` field is what is populated by the calls I removed in this PR.

```
func LateInitializeInstance(in *manualv1alpha1.InstanceParameters, instance *types.Instance, attributes *ec2.DescribeInstanceAttributeOutput)
```

However, the attributes that are removed in my PR aren't actually pulled from `attributes` and instead are populated from the `DescribeInstance` call.

Snippet:
```
	in.EBSOptimized = pointer.LateInitialize(in.EBSOptimized, instance.EbsOptimized)
	in.KernelID = pointer.LateInitialize(in.KernelID, instance.KernelId)
	in.RAMDiskID = pointer.LateInitialize(in.RAMDiskID, instance.RamdiskId)
```

The only one that is not is `InstanceType` which this PR swaps over to use the `DescribeInstances` output.

Reply:
Yeah that makes sense. I guess I mixed up those two.